FirebaseAuth/Sources/AuthProvider/OAuth/FIROAuthProvider.m

```diff
@@ -272,19 +284,25 @@ - (void)getHeadFulLiteURLWithEventID:(NSString *)eventID
                                      }
                                      __strong __typeof__(self) strongSelf = weakSelf;
                                      NSString *bundleID = [NSBundle mainBundle].bundleIdentifier;
-                                     NSString *clienID = strongSelf->_auth.app.options.clientID;
+                                     NSString *clientID = strongSelf->_auth.app.options.clientID;
+                                     NSString *firebaseAppID =
+                                         strongSelf->_auth.app.options.googleAppID;
                                      NSString *apiKey =
                                          strongSelf->_auth.requestConfiguration.APIKey;
                                      NSMutableDictionary *urlArguments = [@{
                                        @"apiKey" : apiKey,
-                                       @"authType" : @"signInWithRedirect",
+                                       @"authType" : kAuthTypeSignInWithRedirect,
                                        @"ibi" : bundleID ?: @"",
-                                       @"clientId" : clienID,
                                        @"sessionId" : [strongSelf hashforString:sessionID],
                                        @"v" : [FIRAuthBackend authUserAgent],
                                        @"eventId" : eventID,
                                        @"providerId" : strongSelf->_providerID,
                                      } mutableCopy];
+                                     if (clientID) {
+                                       urlArguments[@"clientId"] = clientID;
+                                     } else {
+                                       urlArguments[@"appId"] = firebaseAppID;
+                                     }
```

Comment:
I'm not sure if there is a case that someone depends on the "clientId" explicitly. (You should double check FirebaseUI) While this change looks a bit dangerous.

Reply:
I actually suggest keeping the key name "clientId". While it's your call.

Reply:
Do you mean for the URL query parameter? On the web widget side, I have a change in progress for adding support to the appId query parameter (as the logic will branch based on whether the appId or the clientId is passed).

Regarding FirebaseUI, it looks like FUIGoogleAuth depends on the clientId. However, it should be noted that for now this change is a no-op (since the client ID is still being provided in the plist). I will update FirebaseUI in another PR.

Does this address your concerns?

Reply:
Yeah, sounds reasonable. But if we keep using "clientId", you can save the efforts of making the additional changes right?

Reply:
Ah I see, so the web widget needs to send a different parameter in the request to the backend for validation based on if it's the client ID or the app ID, so I don't think the query param can be reused.

Reply:
Got it. Thanks for clarifying!